content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md

```diff
@@ -30,165 +30,318 @@ api_metadata:
+  ResourceClaims for a workload. ResourceClaimTemplates provide Pods with
+  access to separate, similar resources. Each ResourceClaim that Kubernetes
+  generates from the template is bound to a specific Pod. When the Pod
+  terminates, Kubernetes deletes the corresponding ResourceClaim.
 
 ResourceSlice
-: Used by DRA drivers to publish information about resources (typically devices)
-  that are available in the cluster.
-
-DeviceTaintRule
-: Used by admins or control plane components to add device taints
-  to the devices described in ResourceSlices.
-
-All parameters that select devices are defined in the ResourceClaim and
-DeviceClass with in-tree types. Configuration parameters can be embedded there.
-Which configuration parameters are valid depends on the DRA driver -- Kubernetes
-only passes them through without interpreting them.
-
-The `core/v1` `PodSpec` defines ResourceClaims that are needed for a Pod in a
-`resourceClaims` field. Entries in that list reference either a ResourceClaim
-or a ResourceClaimTemplate. When referencing a ResourceClaim, all Pods using
-this PodSpec (for example, inside a Deployment or StatefulSet) share the same
-ResourceClaim instance. When referencing a ResourceClaimTemplate, each Pod gets
-its own instance.
-
-The `resources.claims` list for container resources defines whether a container gets
-access to these resource instances, which makes it possible to share resources
-between one or more containers.
-
-Here is an example for a fictional resource driver. Two ResourceClaim objects
-will get created for this Pod and each container gets access to one of them.
+: Represents one or more resources that are attached to nodes, such as devices.
+  Drivers create and manage ResourceSlices in the cluster. When a ResourceClaim
+  is created and used in a Pod, Kubernetes uses ResourceSlices to find nodes
+  that have access to the claimed resources. Kubernetes allocates resources to
+  the ResourceClaim and schedules the Pod onto a node that can access the
+  resources.
+
+### DeviceClass {#deviceclass}
+  
+A DeviceClass lets cluster admins or device drivers define categories of devices
+in the cluster. DeviceClasses tell operators what devices they can request and
+how they can request those devices. You can use
+[common expression language (CEL)](https://cel.dev) to select devices based on
+specific attributes. A ResourceClaim that references the DeviceClass can then
+request specific configurations within the DeviceClass.
+
+To create a DeviceClass, see
+[Set Up DRA in a Cluster](/docs/tasks/configure-pod-container/assign-resources/set-up-dra-cluster).
+
+### ResourceClaims and ResourceClaimTemplates {#resourceclaims-templates}
+
+A ResourceClaim defines a request for resource allocation. Every ResourceClaim
+references a specific DeviceClass and uses _requests_ and _constraints_ to
+filter the resources in the DeviceClass. ResourceClaims can be created by
```

Comment:
```suggestion
A ResourceClaim defines a request for resource allocation. Every request in a ResourceClaim
references a specific DeviceClass and uses and _selectors_ to
filter the resources in the DeviceClass. ResourceClaims can be created by
```

"selectors" filter devices. "constraints" limit the set of devices. I'm not sure (yet) whether this also mixed up elsewhere. Also note that device classed are per request, not per claim.